[VBLOCKS-4696] E2E tests for real-time transcription

luisrivas opened the pull request:
## Pull Request Details

### Description
- Created integration tests for live transcription functionality.
- Created unit tests for live transcription signaling handler.
- Introduced `isLiveTranscriptionEnabled` property in the Room class to indicate if the live transcription is enabled.
- Updated request handling in tests to properly encode nested objects.

## Burndown

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [x] Updated affected documentation
* [x] Verified locally with `npm test`
* [x] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

luisrivas commented on lib/signaling/v2/livetranscriptionsignaling.js:
```diff
@@ -12,18 +13,25 @@ class LiveTranscriptionSignaling extends MediaSignaling {
       transport.on('message', message => {
         switch (message.type) {
           case 'extension_transcriptions':
-            // TODO: Test and update the message format once this is ready on the backend
```
@charliesantos Let me know if you were thinking of doing any transformation to the message before bubbling it up

charliesantos replied:
With the current implementation, what is the JSON output below?
```javascript
room.on('liveTranscription', console.log);
```

luisrivas replied:
Okay, on it.

luisrivas commented on test/lib/post.js:
```diff
@@ -41,7 +41,11 @@ function request(config, data) {
     request.once('error', reject);
     if (data) {
-      request.write(Object.keys(data).map(key => `${key}=${data[key]}`).join('&'));
+      request.write(
+        Object.entries(data)
+          .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(typeof value === 'object' ? JSON.stringify(value) : value)}`)
+          .join('&')
+      );
```
This is a simple adjustment to support sending requests with nested objects.

charliesantos commented on lib/room.js:
```diff
@@ -636,6 +643,8 @@ function handleSignalingEvents(room, signaling) {
 
   signaling.on('dominantSpeakerChanged', () => room.emit('dominantSpeakerChanged', room.dominantSpeaker));
 
+  signaling.on('liveTranscription', data => room.emit('liveTranscription', data));
```
Thanks 😃  I missed adding this in my initial PR.

luisrivas replied:
Happy to help!

charliesantos commented on lib/signaling/v2/livetranscriptionsignaling.js:
```diff
@@ -4,6 +4,7 @@ const MediaSignaling = require('./mediasignaling');
 
 /**
  * @emits LiveTranscriptionSignaling#transcription
+ * @property {boolean} isEnabled - Whether the connection is live and ready to receive transcription events
```
Why do we need this?

luisrivas replied:
Since `isLiveTranscriptionEnabled` no longer exists, I removed it as well

charliesantos reviewed (changes requested):
I gave it a first pass. I'll continue reviewing the tests in a bit/tomorrow.

luisrivas commented on lib/room.js:
```diff
@@ -16,6 +16,8 @@ let nInstances = 0;
  * @property {?RemoteParticipant} dominantSpeaker - The Dominant Speaker in the
  *   {@link Room}, if any
+ * @property {boolean} isLiveTranscriptionEnabled - Whether or not live transcription is enabled
+ *   in the {@link Room}
```
@charliesantos I removed this property now that we agreed it was no longer necessary

luisrivas commented:
> @luisrivas do we know why we have a lot of tests failing?

No, LiveTranscription tests should fail on stage until the backend supports it, but the rest of the test should pass. I will rerun the tests.

charliesantos commented on test/integration/spec/livetranscription.js:
Can we add tests for
- Should not receive transcription if ConnectOptions.enableLiveTranscription is set to false

luisrivas replied:
Done

luisrivas commented:
> Can we also add unit tests for
> 
> * Under TwilioConnectionTransport, setting enableLiveTranscription flag sets/unsets the MSP
> 
> Also add tsdef tests in twilio-video-tests.ts. And update types.d.ts. I'm not sure if this is part of the current ticket. If not, please submit a new JIRA ticket. Or maybe it's part of the update doc task?

Sure, I have added the unit tests for the TwilioConnectionTransport and also updated the types 👍

charliesantos commented:
> Can we also add unit tests for
> 
> * Under TwilioConnectionTransport, setting enableLiveTranscription flag sets/unsets the MSP
> 
> Also add tsdef tests in twilio-video-tests.ts. And update types.d.ts. I'm not sure if this is part of the current ticket. If not, please submit a new JIRA ticket. Or maybe it's part of the update doc task?

I think you missed adding `ConnectOptions.enableLiveTranscription` tsdef tests (twilio-video-tests.ts) and types.d.ts.

charliesantos commented:
> > Can we also add unit tests for
> > 
> > * Under TwilioConnectionTransport, setting enableLiveTranscription flag sets/unsets the MSP
> > 
> > Also add tsdef tests in twilio-video-tests.ts. And update types.d.ts. I'm not sure if this is part of the current ticket. If not, please submit a new JIRA ticket. Or maybe it's part of the update doc task?
> 
> Sure, I have added the unit tests for the TwilioConnectionTransport and also updated the types 👍

I think you missed adding `ConnectOptions.enableLiveTranscription` tsdef tests (twilio-video-tests.ts) and types.d.ts.

luisrivas commented:
> 
> I think you missed adding `ConnectOptions.enableLiveTranscription` tsdef tests (twilio-video-tests.ts) and types.d.ts.

Good catch! I just added it
